android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java

```diff
@@ -127,4 +278,220 @@ else if (seconds > 60) {
+
+                    Bitmap newBitmap = Bitmap.createBitmap(imageBitmap, startX, startY, layoutWidth, (int) layoutHeight);
+                    Bitmap bitmapWithGradient = ImageUtils.addGradient(mTab.getActivity(), newBitmap, dpToPx(mTab.getActivity(),BOTTOM_TOOLBAR_HEIGHT));
+                    Bitmap offsetBitmap = ImageUtils.topOffset(bitmapWithGradient, dpToPx(mTab.getActivity(),BOTTOM_TOOLBAR_HEIGHT));
+
+                    imageBitmap.recycle();
+                    newBitmap.recycle();
+                    bitmapWithGradient.recycle();
+
+                    // Center vertically, and crop to new center
+                    imageDrawable = new BitmapDrawable(mNewTabPageLayout.getResources(), offsetBitmap);
+                    mNewTabPageLayout.setBackground(imageDrawable);
+
+                    mNewTabPageLayout.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                }
+            });
+        }
+    }
+
+    private void checkForNonDistruptiveBanner(NTPImage ntpImage) {
+        int brOption = NTPUtil.checkForNonDistruptiveBanner(ntpImage, sponsoredTab);
+        if (SponsoredImageUtil.BR_INVALID_OPTION != brOption) {
+            NTPUtil.showNonDistruptiveBanner(mTab.getActivity(), mNewTabPageLayout, brOption, sponsoredTab, newTabListener);
+        }
+    }
+
+    private void checkAndShowNTPImage() {
+        NTPImage ntpImage = sponsoredTab.getTabNTPImage();
+        if(ntpImage == null) {
+            sponsoredTab.setNTPImage(SponsoredImageUtil.getBackgroundImage());
+        } else if (ntpImage instanceof SponsoredImage) {
+            String countryCode = LocaleUtils.getCountryCode();
+            SponsoredImage sponsoredImage = (SponsoredImage) ntpImage;
+            File imageFile = new File(PathUtils.getDataDirectory(), countryCode + "_" + sponsoredImage.getImageUrl());
+            if(!imageFile.exists()) {
+                sponsoredTab.setNTPImage(SponsoredImageUtil.getBackgroundImage());
+                ntpImage = sponsoredTab.getTabNTPImage();
+            }
+        }
+        checkForNonDistruptiveBanner(ntpImage);
+        showNTPImage(ntpImage);
+    }
+
+    private TabObserver mTabObserver = new EmptyTabObserver() {
+        @Override
+        public void onInteractabilityChanged(boolean interactable) {
+            // Force a layout update if the tab is now in the foreground.
+            if (interactable) {
+                checkAndShowNTPImage();
+            } else {
+                if(!isFromBottomSheet){
+                    mNewTabPageLayout.setBackgroundResource(0);
+                    if (imageDrawable != null && imageDrawable.getBitmap() != null && !imageDrawable.getBitmap().isRecycled()) {
+                        imageDrawable.getBitmap().recycle();
+                    }
+                }
+            }
+        }
+    };
+
+    private NewTabListener newTabListener = new NewTabListener() {
```

Comment:
is this supposed to be `NewTabPageListener`? Because `NewTabListener` doesn't seem to apply

Reply:
i agree. i should change the name.